Gems/ROS2/Code/Source/Frame/ROS2Transform.cpp

```diff
@@ -19,6 +20,20 @@ namespace ROS2
         , m_childFrame(AZStd::move(childFrame))
         , m_isDynamic(isDynamic)
     {
+        auto splitName = ROS2Names::SplitIntoNamespaceAndName(m_childFrame);
+        if (splitName.first.empty())
+        {   // No namespace in child frame, no need for a publisher.
+            return;
+        }
+
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+        rclcpp::QoS qos = tf2_ros::StaticBroadcasterQoS();
+        if (m_isDynamic)
+        {
+            qos = tf2_ros::DynamicBroadcasterQoS();
+        }
+        AZStd::string fullTopic = ROS2Names::GetNamespacedName(splitName.first, m_isDynamic ? "tf" : "tf_static");
+        m_transformPublisher = ros2Node->create_publisher<geometry_msgs::msg::TransformStamped>(fullTopic.data(), qos);
```

Comment:
Using `geometry_msgs::msg::TransformStamped` over `tf2_msgs::msg::TFMessage` prevents rqt and nav2 from detecting tfs correctly.

Reply:
You are correct. Static broadcaster publishes a vector of TransformStamped, but (dynamic) broadcaster publishes TFMessage, see here:
https://github.com/ros2/geometry2/blob/rolling/tf2_ros/src/transform_broadcaster.cpp.

We should template the ROS2Transform class with the message type.